smithy-model/src/main/java/software/amazon/smithy/model/validation/validators/PaginatedTraitValidator.java

```diff
@@ -164,9 +182,53 @@
 
         abstract Set<ShapeType> validTargets();
 
-        abstract Optional<String> getMemberName(OperationIndex index, OperationShape operation, PaginatedTrait trait);
+        abstract Optional<String> getMemberPath(OperationIndex opIndex, OperationShape operation, PaginatedTrait trait);
+
+        abstract Optional<MemberShape> getMember(
+                ShapeIndex index, OperationIndex opIndex, OperationShape operation, PaginatedTrait trait
+        );
+
+        boolean pathsAllowed() {
+            return false;
+        }
+    }
+
+    private abstract static class OutputPropertyValidator extends PropertyValidator {
+
+        @Override
+        boolean pathsAllowed() {
+            return true;
+        }
+
+        Optional<MemberShape> getMember(
+                ShapeIndex index, OperationIndex opIndex, OperationShape operation, PaginatedTrait trait
+        ) {
+            // Split up the path expression into a list of member names
+            Optional<List<String>> memberNames = getMemberPath(opIndex, operation, trait)
```

Comment:
Minor: this could be simplified to:

```java
Optional<List<String>> memberNames = getMemberPath(opIndex, operation, trait)
        .map(value -> Arrays.asList(value.split("\\.")))
        .orElse(Collections.emptyList());
```